### PR TITLE
Add version expression property to luisadapter and make options expression properties 

### LINKED ISSUE #4534: LuisRecgonizer needs ability to bind Version to settings
## Version

4.10.3

## Describe the bug

Luis models have AppId/Slot or AppId/Version.  The AppId/Version can be hard coded, but can't be bund to settings.  For LUBuild to work correctly in production it needs to be bindable to both AppId and Version in settings.

This allows

```javascript
{
   "$kind":"Microsoft.LuisRecognizer",
   "applicationId":"=settings.foo.lu.applicationId",
   "version":"=settings.foo.lu.version"
   ...
}
```

Without this change we would have to rewrite every dialog file on build to target fixed version.

> NOTE: recommend we take this for next 4.10.X release.

### PR DESCRIPTION
**Pull request by tomlm**
Fixes #4534

## Description

Add Version expression property and options which use ExpressionPRoperties so they can be bound to settings.

## Specific Changes

*   Adds Version expression property to LuisAdaptiveRecognizer.
*   Adds LuisAdaptivePredictionOptions which uses ExpressionProperties

### CODE REVIEW

**Comment by chrimc62** [comment]
>                     "description": "LUIS application version to use."

Wouldn't a better change be to turn these into expressions?

---
Refers to: libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema:116 in 5d9c029. [](commit_id = 5d9c02982b3948bd7f9531ca61d9ac5c9b582c49, deletion_comment = False)

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema [hunk 1]
```diff
@@ -81,34 +86,34 @@
             "description": "Options to control LUIS prediction behavior.",
             "properties": {
                 "includeAllIntents": {
+                    "$ref": "schema:#/definitions/booleanExpression",
                     "type": "boolean",
                     "title": "Include all intents",
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema:91]: You should probably remove all of these types--you get them from the included expression.

**Review by chrimc62 — APPROVED** [review]
:shipit:

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema (hunk 1 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/Schemas/Microsoft.LuisRecognizer.schema:90]: removed

**Pull request merged**